dace/sdfg/nodes.py

```diff
@@ -377,6 +386,24 @@ def validate(self, sdfg, state):
     def free_symbols(self) -> Set[str]:
         return self.code.get_free_symbols(self.in_connectors.keys() | self.out_connectors.keys())
 
+    def has_side_effects(self, sdfg) -> bool:
+        """
+        Returns True if this tasklet may have other side effects (e.g., calling stateful libraries, communicating).
+        """
+        # If side effects property is set, takes precedence over node analysis
+        if self.side_effects is not None:
+            return self.side_effects
+
+        # If side effect property is not defined, find calls within tasklet
+        if self.code.language == dace.dtypes.Language.Python and self.code.code:
+            for stmt in self.code.code:
+                for n in ast.walk(stmt):
+                    if isinstance(n, ast.Call):
+                        cname = rname(n.func)
+                        if cname in sdfg.symbols or cname in sdfg.arrays:
```

Comment:
If the function call name is one of the symbols or data containers it has side effects? Might need a doc comment, or am I reading it wrong?

Reply:
the check basically asks "is this function a callback" (they may be `Scalar(dace.callback)` or symbol typed `dace.callback`. Will comment.